python/seldon_core/seldon_methods.py

```diff
@@ -29,6 +30,27 @@
 logger = logging.getLogger(__name__)
 
 
+def handle_raw_custom_metrics(
+    msg: Union[prediction_pb2.SeldonMessage, Dict],
+    seldon_metrics: SeldonMetrics,
+    is_proto: bool,
+):
+    """
+    Update SeldonMetrics object with custom metrics from raw methods.
+    If INCLUDE_METRICS_IN_CLIENT_RESPONSE environmental variable is set to "true"
+    metrics will be dropped from msg.
+    """
+    if is_proto:
+        metrics = seldon_message_to_json(msg.meta).get("metrics", [])
+        if metrics and not return_metrics_to_client():
+            del msg.meta.metrics[:]
+    else:
+        metrics = msg.get("meta", {}).get("metrics", [])
+        if metrics and not return_metrics_to_client():
+            del msg["meta"]["metrics"]
```

Comment:
You are adding a guard above (ie `.get(...)` but not here, which would lead into a KeyError if it's not there

Reply:
I had the feeling it will be handled by `if metrics` part of conditional statement.